WordPress/Classes/Models/AbstractPost.m

```diff
@@ -452,8 +452,17 @@ - (BOOL)isDraft
 
 - (void)publishImmediately
 {
-    self.dateModified = [NSDate date];
-    [self setDateCreated:self.dateModified];
+    // If the post was modified later than it was created
+    // then this is a post that was published at some point
+    // we will not update the dateCreated
```

Comment:
Well, it will publish the post immediately, it just won't update the `create_date` which is actually the `publish_date` when the post has been published before. 
I'm not crazy about the name either, but couldn't come up with a better one myself. Any ideas?